--- ldap_helper_new.php
+++ ldap_helper.php
@@ -27,7 +27,11 @@
     } else {
         $ds = false;
         if ($ldap_server[$server_id]['protoversion'] == 'ldapv3' && $ldap_server[$server_id]['encrypt'] != 'ldaps') {
-            $ds = ldap_connect($ldap_server[$server_id]['server'], $ldap_server[$server_id]['port']);
+            if ($ldap_server[$server_id]['encrypt'] == 'ldaps') {
+                $ds = ldap_connect("ldaps://".$ldap_server[$server_id]['server'], $ldap_server[$server_id]['port']);
+            } else {
+                $ds = ldap_connect($ldap_server[$server_id]['server'], $ldap_server[$server_id]['port']);
+            }
             ldap_set_option($ds, LDAP_OPT_PROTOCOL_VERSION, 3);

             if (!$ldap_server[$server_id]['referrals']) {
