diff --git a/application/views/admin/survey/topbar/token_bar.php b/application/views/admin/survey/topbar/token_bar.php
index 922873c76f..fc905490ed 100644
--- a/application/views/admin/survey/topbar/token_bar.php
+++ b/application/views/admin/survey/topbar/token_bar.php
@@ -211,7 +211,7 @@ if (!$onlyclose) {
             $isDisabled = true;
         }
 
-        if (!($oSurvey->bounceprocessing != 'N' || ($oSurvey->bounceprocessing == 'G' && getGlobalSetting('bounceaccounttype') != 'off'))) {
+        if (($oSurvey->bounceprocessing == 'N' || ($oSurvey->bounceprocessing == 'G' && getGlobalSetting('bounceaccounttype') == 'off'))) {
             $errorMessages[] = gT("Bounce processing is deactivated either application-wide or for this survey in particular.");
             $isDisabled = true;
         }
@@ -219,11 +219,10 @@ if (!$onlyclose) {
         $errorMessage = join("\n", $errorMessages);
 
         $buttons['bounce_processing'] = [
-            'url' => '#',
+            'url' => $isDisabled ? '' : $this->createUrl("admin/tokens/sa/bounceprocessing/surveyid/$sid"),
             'alerttext' => $errorMessage,
             'icon' => 'fa fa-cogs',
             'id' => 'startbounceprocessing',
-            'data-url' => $isDisabled ? '' : $this->createUrl("admin/tokens/sa/bounceprocessing/surveyid/$sid"),
             'name' => gT('Start bounce processing'),
             'class' => 'pjax ' . ($isDisabled ? ' disabled' : ' '),
         ];
@@ -276,7 +275,7 @@ if (!$onlyclose) {
 
     ###### Right aligned
 
-    
+
     if ( Yii::app()->request->getParam('saveButton', 'false')!='true' && (Permission::model()->hasSurveyPermission($sid, 'surveysettings', 'update')
         || Permission::model()->hasSurveyPermission($sid, 'tokens', 'delete'))
     ) {
