--- token_bar.php.original	2020-07-24 10:35:43.540889910 +0200
+++ token_bar.php	2020-07-24 10:36:18.976259838 +0200
@@ -211,7 +211,7 @@
             $isDisabled = true;
         }
 
-        if (($oSurvey->bounceprocessing != 'N' || ($oSurvey->bounceprocessing == 'G' && getGlobalSetting('bounceaccounttype') != 'off'))) {
+        if (($oSurvey->bounceprocessing == 'N' || ($oSurvey->bounceprocessing == 'G' && getGlobalSetting('bounceaccounttype') == 'off'))) {
             $errorMessages[] = gT("Bounce processing is deactivated either application-wide or for this survey in particular.");
             $isDisabled = true;
         }
@@ -276,7 +276,7 @@
 
     ###### Right aligned
 
-    
+
     if ( Yii::app()->request->getParam('saveButton', 'false')!='true' && (Permission::model()->hasSurveyPermission($sid, 'surveysettings', 'update')
         || Permission::model()->hasSurveyPermission($sid, 'tokens', 'delete'))
     ) {
